### PR TITLE
5921 implement sonarqube enhancements

### PR DESCRIPTION
**Pull request by AnumehaSrivastava05**

### CODE REVIEW

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java [hunk 1]
```diff
@@ -76,34 +80,50 @@ public static FHIRJsonMessage parseJson2FHIRMsg(final JSONObject jsonObj) {
...
 			if (value instanceof String) {
-				if (key.equals(FHIR_JSONMESSAGE_RESOURCETYPE)) {
-					fhirJsonMsg.setResourceType(value.toString());
-				} else if (key.equals(FHIR_JSONMESSAGE_STATUS)) {
-					fhirJsonMsg.setStatus(value.toString());
-				} else {
-					logger.error("This is not an valid FHIR message!");
-					return null;
-				}
+				parseJsonString(fhirJsonMsg, key, value);
```
Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java:83]: Business logic in new method is not same as old logic.
'return null' is ignored.

Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java:83]: why pass 2 params when you can pass 'hs' object

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java [hunk 2]
```diff
@@ -76,34 +80,50 @@ public static FHIRJsonMessage parseJson2FHIRMsg(final JSONObject jsonObj) {
...
-						logger.error("This is not an valid FHIR message!");
-						return null;
-					}
-				}
+			else if (key.equals(FHIR_JSON_MESSAGE_CONTENT) && value instanceof JSONArray) {
+				parseJsonArray(fhirJsonMsg, value);
 
-			}
+			} else 
+				return null;
```
Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java:89]: This is not same as 'return null' in old business logic.

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java [hunk 3]
```diff
@@ -76,34 +80,50 @@ public static FHIRJsonMessage parseJson2FHIRMsg(final JSONObject jsonObj) {
...
+
+		for (Entry<String, Object> attach : attachmentJson.entrySet()) {
+			String attachKey = attach.getKey();
+			Object attachValue = attach.getValue();
+			if (attachKey.equals(FHIR_JSON_MESSAGE_TYPE)) {
+				fhirJsonMsg.setContentType(attachValue.toString());
+			} else if (attachKey.equals(FHIR_JSON_MESSAGE_DATA)) {
+				fhirJsonMsg.setV2MessageData(attachValue.toString());
+			} else {
+				logger.error("This is not an valid FHIR message!");
```
Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java:110]: Error handling is not complete

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java [hunk 4]
```diff
@@ -76,34 +80,50 @@ public static FHIRJsonMessage parseJson2FHIRMsg(final JSONObject jsonObj) {
...
+	 * @param key
+	 * @param value
+	 */
+	private static void parseJsonString(FHIRJsonMessage fhirJsonMsg, String key, Object value) {
+		if (key.equals(FHIR_JSON_MESSAGE_RESOURCETYPE)) {
+			fhirJsonMsg.setResourceType(value.toString());
+		} else if (key.equals(FHIR_JSON_MESSAGE_STATUS)) {
+			fhirJsonMsg.setStatus(value.toString());
+		} else {
+			logger.error("This is not an valid FHIR message!");
```
Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java:126]: Error handling is not complete

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/MessageUtil.java [hunk 5]
```diff
@@ -5,7 +5,10 @@
 
 public class MessageUtil {
 	
-	public static final Map<String, String> mTypeCollection = new HashMap<String, String>();
+	public static final Map<String, String> mTypeCollection = new HashMap<>();//NOSONAR
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/MessageUtil.java:8]: Whats this comment for?

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java [hunk 6]
```diff
@@ -38,10 +38,10 @@ public void populateJMSRequestHeader(Exchange exchange, String v2Message) throws
 		} catch (CSIException e) {
 			logger.error("{} - TransactionId: {}, Exception while converting hexadecimal message Control Id to byte {}",
 				methodName, exchange.getExchangeId(), e.getMessage());
-        	throw e;
+        	throw new IllegalArgumentException("Exception while converting hexadecimal message Control Id to byte");
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java:41]: This should be a constant error message

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java [hunk 7]
```diff
@@ -38,10 +38,10 @@ public void populateJMSRequestHeader(Exchange exchange, String v2Message) throws
...
 		}
 		
-		String exchangeId = exchange.getExchangeId().replaceAll("-", "");
+		String exchangeId = exchange.getExchangeId().replace("-", "");
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java:44]: This changes the business logic. Is this required?

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateVersionInformation.java [hunk 8]
```diff
@@ -29,14 +29,15 @@ public static JSONObject getVersionInformation() {
 		// init a JSON object
 		JSONObject v2JsonObj = new JSONObject();
 		v2JsonObj.put(IMPLEMENTATION_VERSION_KEY, pck.getImplementationVersion());
-		logger.debug("{} - The JSON Message is: {}", methodName, v2JsonObj.toJSONString());
+		String jsonString = v2JsonObj.toJSONString();
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateVersionInformation.java:32]: why new variable is introduced?

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java [hunk 9]
```diff
@@ -69,6 +69,10 @@
 	
 	public static final String PROTOCOL_HTTP = "HTTP";
 	public static final String PROTOCOL_MQ = "MQ";	
+	
+	private Util() {
+		throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java:74]: Error message Should be a constant

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java [hunk 10]
```diff
@@ -21,12 +21,15 @@
...
 public class HIBCRoute extends BaseRoute {
 
+	private static final String DIRECT_AUDIT = "direct:audit";
+	private static final String DIRECT_HIBC_MQ = "direct:hibcMQ";
+
 	@Override
 	public void configure() throws Exception {
 
-		String hibcHttpUrl = String.format(properties.getValue(ApplicationProperty.HIBC_HTTP_URI) + "?bridgeEndpoint=true");
+		String hibcHttpUrl = String.format(properties.getValue(ApplicationProperty.HIBC_HTTP_URI) + "?bridgeEndpoint=true");//NOSONAR
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java:30]: is this comment to ignore sonar code smell?

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java [hunk 11]
```diff
@@ -51,37 +54,37 @@ public void configure() throws Exception {
...
 	     	.setBody().method(new Base64Encoder()).id("HIBCBase64Encoder")
             .setBody().method(new ProcessV2ToJson()).id("HIBCProcessV2ToJson")
 	     	.setHeader(Exchange.CONTENT_TYPE, constant("application/json"))
 	        .setHeader("CamelHttpMethod", POST)
 	     	.to("log:HttpLogger?level=INFO&showBody=true&showHeaders=true&multiline=true")
 	     	.to(hibcHttpUrl).id("ToHibcHttpUrl")
 	     	.log("Received response from HIBC")
-	     	.setBody().method(new FhirPayloadExtractor())
+	     	.bean(FhirPayloadExtractor.class).log("Decoded V2: ${body}")
 	     	.log("Decoded V2: ${body}")
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java:66]: logging twice now

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/util/LoggingUtil.java [hunk 12]
```diff
@@ -6,7 +6,11 @@
  * Utility class that contains methods related to logging.
  * 
  */
-public class LoggingUtil {
+public final class LoggingUtil {
+	
+	private LoggingUtil() {
+		throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
```
Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/util/LoggingUtil.java:12]: Error message should be a constant.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/JMBRoute.java [hunk 13]
```diff
@@ -50,37 +53,37 @@ public void configure() throws Exception {
...
 	     	.setBody().method(new Base64Encoder()).id("JMBBase64Encoder")
             .setBody().method(new ProcessV2ToJson()).id("JMBProcessV2ToJson")
 	     	.setHeader(Exchange.CONTENT_TYPE, constant("application/json"))
 	        .setHeader("CamelHttpMethod", POST)
 	     	.to("log:HttpLogger?level=INFO&showBody=true&showHeaders=true&multiline=true")
 	     	.to(jmbHttpUrl).id("ToJmbHttpUrl")
 	     	.log("Received response from JMB for R32")
-	     	.setBody().method(new FhirPayloadExtractor())
+	     	.bean(FhirPayloadExtractor.class).log("Decoded V2: ${body}")
 	     	.log("Decoded V2: ${body}")
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/JMBRoute.java:65]: logging twice

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java (hunk 1 above)
Comment (AnumehaSrivastava05) [reply hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java]: Passed object

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/MessageUtil.java (hunk 5 above)
Comment (AnumehaSrivastava05) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/MessageUtil.java]: https://rules.sonarsource.com/java/RSPEC-1104

Added getter.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java (hunk 6 above)
Comment (AnumehaSrivastava05) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java]: Added Constant

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java (hunk 7 above)
Comment (AnumehaSrivastava05) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java:45]: https://rules.sonarsource.com/java/RSPEC-5361

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java (hunk 9 above)
Comment (AnumehaSrivastava05) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java]: Added constant

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java (hunk 10 above)
Comment (AnumehaSrivastava05) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java:30]: Yes.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java (hunk 11 above)
Comment (AnumehaSrivastava05) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java:66]: Removed.

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/util/LoggingUtil.java (hunk 12 above)
Comment (AnumehaSrivastava05) [reply hn-common/src/main/java/ca/bc/gov/hlth/hncommon/util/LoggingUtil.java]: Added Constant.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java [hunk 14]
```diff
@@ -38,7 +39,7 @@ public void populateJMSRequestHeader(Exchange exchange, String v2Message) throws
 		} catch (CSIException e) {
 			logger.error("{} - TransactionId: {}, Exception while converting hexadecimal message Control Id to byte {}",
 				methodName, exchange.getExchangeId(), e.getMessage());
-        	throw new IllegalArgumentException("Exception while converting hexadecimal message Control Id to byte");
+        	throw new IllegalArgumentException(HEXA_TO_BYTE_ERROR);
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java:42]: Should not this be part of ErrorMessage.java?

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java [hunk 15]
```diff
@@ -108,22 +114,29 @@ private static void parseJsonArray(FHIRJsonMessage fhirJsonMsg, Object value) {
 				fhirJsonMsg.setV2MessageData(attachValue.toString());
 			} else {
 				logger.error("This is not an valid FHIR message!");
+				isParsed = false;
```
Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java:117]: Business logic is not same as old code. This change will process the for loop before returning false. In old business logic, it returned null first time only when an error is logged.

File: hn-common/src/main/java/ca/bc/gov/hlth/hncommon/util/LoggingUtil.java [hunk 16]
```diff
@@ -8,8 +8,10 @@
  */
 public final class LoggingUtil {
 	
+	private static final String NO_UTIL_INSTANCE = "This is a utility class and cannot be instantiated";
+
 	private LoggingUtil() {
-		throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
+		throw new UnsupportedOperationException(NO_UTIL_INSTANCE);
```
Comment (pankaj-cgi) [inline hn-common/src/main/java/ca/bc/gov/hlth/hncommon/util/LoggingUtil.java:14]: part of ErrorMessage.java?

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java [hunk 17]
```diff
@@ -71,7 +72,7 @@
 	public static final String PROTOCOL_MQ = "MQ";	
 	
 	private Util() {
-		throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
+		throw new UnsupportedOperationException(UTIL_NO_INSTANCE_MESSAGE);
```
Comment (pankaj-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java:75]: same: ErrorMessage.java?

**Review by pankaj-cgi — CHANGES_REQUESTED** [review]

**Review by pankaj-cgi — CHANGES_REQUESTED** [review]

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java (hunk 10 above)
Comment (pankaj-cgi) [reply hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/HIBCRoute.java:30]: Will be better if we can add Sonar ignore errors at one place. If not trivial, adding a comment of what sonar error we are ignoring.

**Review by pankaj-cgi — CHANGES_REQUESTED** [review]

**Review by pankaj-cgi — APPROVED** [review]

**Pull request merged**